packages/arb-ts/src/lib/bridge.ts

```diff
@@ -104,4 +109,258 @@ export class Bridge extends L2Bridge {
+      this.l1Bridge.l1Provider
+    )
+
+    const activeOutbox = await bridge.allowedOutboxList(0)
+    try {
+      // index 1 should not exist
+      await bridge.allowedOutboxList(1)
+      console.error('There is more than 1 outbox registered with the bridge?!')
+    } catch (e) {
+      // this should fail!
+      console.log('All is good')
+    }
+
+    const outboxExecuteTransactionReceipt = await this.tryOutboxExecute(
+      activeOutbox,
+      batchNumber,
+      proof,
+      path,
+      l2Sender,
+      l1Dest,
+      l2Block,
+      l1Block,
+      proofTimestamp,
+      amount,
+      calldataForL1
+    )
+    return outboxExecuteTransactionReceipt
+  }
+
+  public tryOutboxExecute = async (
+    activeOutboxAddress: string,
+    batchNumber: BigNumber,
+    proof: Array<string>,
+    path: BigNumber,
+    l2Sender: string,
+    l1Dest: string,
+    l2Block: BigNumber,
+    l1Block: BigNumber,
+    timestamp: BigNumber,
+    amount: BigNumber,
+    calldataForL1: string,
+    retryDelay = 500
+  ): Promise<ContractReceipt> => {
+    const outbox = OutboxFactory.connect(
+      activeOutboxAddress,
+      this.l1Bridge.l1Signer
+    )
+    await this.waitUntilOutboxEntryCreated(batchNumber, activeOutboxAddress)
+    try {
+      // TODO: wait until assertion is confirmed before execute
+      // We can predict and print number of missing blocks
+      // if not challenged
+      const outboxExecute = await outbox.executeTransaction(
+        batchNumber,
+        proof,
+        path,
+        l2Sender,
+        l1Dest,
+        l2Block,
+        l1Block,
+        timestamp,
+        amount,
+        calldataForL1
+      )
+      console.log(`Transaction hash: ${outboxExecute.hash}`)
+      console.log('Waiting for receipt')
+      const receipt = await outboxExecute.wait()
+      console.log('Receipt emitted')
+      return receipt
+    } catch (e) {
+      console.log('failed to execute tx')
+      console.log(e)
+      console.log('Waiting for delay before retrying')
+      // TODO: should exponential backoff?
+      await this.wait(retryDelay)
+      console.log('Retrying now')
+      return this.tryOutboxExecute(
+        activeOutboxAddress,
+        batchNumber,
+        proof,
+        path,
+        l2Sender,
+        l1Dest,
+        l2Block,
+        l1Block,
+        timestamp,
+        amount,
+        calldataForL1,
+        retryDelay
+      )
+    }
+  }
+
+  public tryGetProof = async (
+    batchNumber: BigNumber,
+    indexInBatch: BigNumber,
+    retryDelay = 500
+  ): Promise<{
+    proof: Array<string>
+    path: BigNumber
+    l2Sender: string
+    l1Dest: string
+    l2Block: BigNumber
+    l1Block: BigNumber
+    timestamp: BigNumber
+    amount: BigNumber
+    calldataForL1: string
+  }> => {
+    const nodeInterfaceAddress = '0x00000000000000000000000000000000000000C8'
+
+    const contractInterface = new ethers.utils.Interface([
+      `function lookupMessageBatchProof(uint256 batchNum, uint64 index)
```

Comment:
Should be able to get the generated ABI for this now that it's been moved to arb-bridge-peripherals

Reply:
I think we should address this and other abi issues in a separate PR so we can play with the generated files/script without creating a monster diff